backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/chip/BpkChip.kt

```diff
@@ -67,120 +44,63 @@ fun BpkChip(
   enabled: Boolean = true,
   style: BpkChipStyle = BpkChipStyle.Default,
   icon: BpkIcon? = null,
-  type: BpkChipType = BpkChipType.Option,
 ) {
+  BpkChipImpl(
+    text = text,
+    selected = selected,
+    enabled = enabled,
+    style = style,
+    icon = icon,
+    type = BpkChipType.Selectable,
+    onSelectedChange = onSelectedChange,
+    modifier = modifier,
+  )
+}
 
-  val interactionSource = remember { MutableInteractionSource() }
-
-  val backgroundColor by animateColorAsState(
-    targetValue = when {
-      !enabled -> BpkChipColors.disabledBackground
-      selected -> style.selectedBackgroundColor
-      else -> interactionSource.animateAsColor(
-        default = style.backgroundColor,
-        pressed = style.pressedBackgroundColor,
-      )
-    },
+@Composable
+fun BpkDropdownChip(
+  text: String,
+  modifier: Modifier = Modifier,
+  selected: Boolean = false,
+  onSelectedChange: ((Boolean) -> Unit)? = null,
+  enabled: Boolean = true,
+  style: BpkChipStyle = BpkChipStyle.Default,
+  icon: BpkIcon? = null,
+) {
+  BpkChipImpl(
+    text = text,
+    selected = selected,
+    enabled = enabled,
+    style = style,
+    icon = icon,
+    type = BpkChipType.Dropdown,
+    onSelectedChange = onSelectedChange,
+    modifier = modifier,
   )
+}
 
-  val contentColor by animateColorAsState(
-    targetValue = when {
-      !enabled -> BpkTheme.colors.textDisabled
-      selected -> style.selectedContentColor
-      else -> interactionSource.animateAsColor(
-        default = style.contentColor,
-        pressed = BpkTheme.colors.textPrimary,
-      )
+@Composable
+fun BpkDismissibleChip(
+  text: String,
+  modifier: Modifier = Modifier,
+  onClick: (() -> Unit)? = null,
+  style: BpkChipStyle = BpkChipStyle.Default,
+  icon: BpkIcon? = null,
+) {
+  val interactionSource = remember { MutableInteractionSource() }
+  BpkChipImpl(
+    text = text,
+    selected = true,
+    enabled = true,
+    style = style,
+    icon = icon,
+    type = BpkChipType.Dismiss,
+    interactionSource = interactionSource,
+    modifier = modifier.applyIf(onClick != null) {
+      clickable(
+        interactionSource = interactionSource,
+        indication = LocalIndication.current,
+      ) { onClick!!.invoke() }
```

Comment:
could this be replaced by 
```kt
modifier = onClick?.let {
   // ... clickable modifier
   // { onClick?.let { it.invoke()) }
}
```

not sure if this works in compose due to how contexts work here

Reply:
not quite! the modifier will need to be applied always - the `applyIf` _might_ be wrongly named, though as it is not the modifier that is not being applied if the condition matches, but the `clickable` part on the modifier